shared/sub_plugin.go

```diff
@@ -0,0 +1,61 @@
+package shared
+
+import (
+	"context"
+
+	"github.com/smartcontractkit/libocr/commontypes"
+)
+
+type AttributedObservation[ObservationType any] struct {
+	OracleID    commontypes.OracleID
+	Observation ObservationType
+}
+
+// SubPlugin This interface is to create multiple SubPlugins/Processors under OCR plugin.
```

Comment:
Let's rename subplugin to `PluginProcessor` and get rid of the ambiguity  of SubPlugins/Processors